pkg/logql/log/fmt_test.go

```diff
@@ -923,3 +924,46 @@ func TestInvalidUnixTimes(t *testing.T) {
 	_, err = unixToTime("464")
 	require.Error(t, err)
 }
+
+func TestMapPoolPanic(_ *testing.T) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	wgFinished := sync.WaitGroup{}
+
+	ls := labels.FromStrings("cluster", "us-central-0")
+	builder := NewBaseLabelsBuilder().ForLabels(ls, ls.Hash())
+	a := newMustLineFormatter(`[1m{{if .level }}{{alignRight 5 .level}}{{else if .severity}}{{alignRight 5 .severity}}{{end}}[0m [90m[{{alignRight 10 .resources_service_instance_id}}{{if .attributes_thread_name}}/{{alignRight 20 .attributes_thread_name}}{{else if eq "java" .resources_telemetry_sdk_language }}                    {{end}}][0m [36m{{if .instrumentation_scope_name }}{{alignRight 40 .instrumentation_scope_name}}{{end}}[0m {{.body}} {{if .traceid}} [37m[3m[traceid={{.traceid}}]{{end}}`)
```

Comment:
Might be helpful to explain here why such an exotic test case is required.

Reply:
This was the only query that I found caused the panic in the first place, though not reliably within smaller environments/smaller time ranges. I tracked it down to the actual panics occurring when we first noticed this, and then reran every query I could find in that time range. This large line format was the only one that reliably caused the panic to occur in larger environments, and in dev/testing with a siginficantly large time range (7-30d) it would happen reliably as well.

Reply:
I meant as a comment, but at least we have it here in the PR for archaeology's sake 👍 thanks